ui/DesignSystem/Component/Timeline/TimelineItem.svelte

```diff
@@ -0,0 +1,132 @@
+<script>
+  import {
+    Avatar,
+    Button,
+    Text,
+    Icon,
+    Input,
+  } from "../../../DesignSystem/Primitive";
+
+  export let item = { variant: "comment" }; // timestamp | open-issue | close-issue | reopen-issue | comment
+  const avatarFallback2 = {
+    background: {
+      r: 122,
+      g: 112,
+      b: 90,
+    },
+    emoji: "💡",
+  };
+</script>
+
+<style>
+  li {
+    display: flex;
+    flex: 1;
+    padding: 0 16px 0 48px;
+    height: 48px;
+    position: relative;
+  }
+  li::before {
+    position: absolute;
+    top: 24px;
+    bottom: 0;
+    left: 59px;
+    display: block;
+    width: 2px;
+    content: "";
+    background-color: var(--color-foreground-level-2);
+  }
+  li:last-child::before {
+    display: none;
+  }
+  .comment {
+    padding-left: 0;
+  }
+  .comment-section {
+    flex: 1;
+  }
+  .comment-meta {
+    display: flex;
+    justify-content: space-between;
+    margin-top: 16px;
+  }
+  .actions {
+    display: flex;
+  }
+</style>
+
+{#if item.variant === 'timestamp'}
```

Comment:
Having these long chains of if else conditions with interleaved markup usually doens't lend it well to ongoing maintenance. Given that our `Timeline` component is pretty plain at the moment it would be the perfect place to the decide what variant to render. So instead of a general-purpose `Item.svelte` component we build one per purpose.

I'd recommend one for `Comment` and one for `State` (which would be used for **open**, **reopen** and **close**).

Reply:
Also that means that the items don't have to render an `li` but that is responsiblilty of the `Timeline`.

Reply:
Yeah, this was in line with what @sarahscott was bringing up, refactored a bit to make more sense.